src/app/folder/folderlist.component.ts

```diff
@@ -220,6 +277,120 @@ export class FolderListComponent {
         ).subscribe(() => this.messagelistservice.refreshFolderCount());
     }
 
+    async folderReorderingDrop(sourceFolderId: number, destinationFolderId: number, aboveOrBelowOrInside: number) {
+        if (sourceFolderId === destinationFolderId) {
+            // can't move a folder above, below or inside itself
+            return;
+        }
+
+        const folders = await this.messagelistservice.folderCountSubject.pipe(take(1)).toPromise();
+        let sourceIndex = folders.findIndex(fld => fld.folderId === sourceFolderId);
+        let destinationIndex = folders.findIndex(folder => folder.folderId === destinationFolderId);
+
+        console.log(`move folder ${folders[sourceIndex].folderPath} ${
+                            aboveOrBelowOrInside === 1 ? 'above' :
+                            aboveOrBelowOrInside === 2 ? 'below' :
+                            aboveOrBelowOrInside === 3 ? 'inside' : ''
+                        } ${folders[destinationIndex].folderPath}`);
+
+        let destinationFolderLevel = 0;
+        let destinationParent = '';
+
+        const getParentFromFolderPath = folderPath => {
+            const pathArr = folderPath.split('/');
+            return pathArr.slice(0, pathArr.length - 1).join('/');
+        };
+
+        let moveCount = 1;
+        while ( sourceIndex + moveCount < folders.length &&
+                folders[sourceIndex + moveCount].folderLevel > folders[sourceIndex].folderLevel) {
+            // also move all sub folders if any
+            moveCount ++;
+        }
+
+        switch (aboveOrBelowOrInside) {
+            case 1:
```

Comment:
Not using enums either.

Reply:
ohh... I hope I've corrected all of those now :-) Thanks for finding these!